### PR TITLE
fix broken html tags during translation

### LINKED ISSUE #3: [Bug]: HTML translations don't preserve tags
### What happened?

When translating html text, especially text that is over the limits of each driver and gets split and ran through `translateBatch`, the strings are sometimes split in the middle of words or in the middle of html tags. And while generally, the drivers preserve html tags and avoid translating them, these splits result in broken tags and incorrect translations.

Additionally, the amazon driver specifically seems to move some paragraph ending tags and often pushes punctuation out of the paragraph.

### How to reproduce the bug

- Attempt to translate a big amount of html, that will get split into chunks.
- Observe that the html tags are not preserved
- Even small amounts of html text can result in shifted or moved tags when using the amazon driver

### Package Version

0.0.0

### PHP Version

8.3.0

### Laravel Version

9.0.0

### Which operating systems does with happen with?

_No response_

### Notes

_No response_

### PR DESCRIPTION
**Pull request by a-drew**
## Summary

- Fixes issues with long text where incorrect splits result in html tags being treated as actual text to translate
- Adds better html support for AmazonTranslate by using its real-time document translation feature

## Testing

- Copy over some html content that is about 30-50k in length
- run str_split and run html_split on that content, use limit 10000, to see how str_split sometimes cuts html tags in two
- Run translate with format html and confirm that the translated content doesn't have broken up html
- Re-test with the amazon driver - the can be seen visually in another project

### CODE REVIEW

File: src/helpers.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        $lastSplit = 0;
+        $lastTagPos = 0;
+        foreach ($endTags as $tag) {
+            $tagPos = $tag[1] + strlen($tag[0]);
+            if ($tagPos - $lastSplit > $maxLength) {
+                $chunks[] = substr($html, $lastSplit, $lastTagPos - $lastSplit);
+                $lastSplit = $lastTagPos;
+            }
+            $lastTagPos = $tagPos;
+        }
```
Comment (m-triassi) [inline src/helpers.php:27]: this logic is a little hard to grok for me, i definitely think it could use some comments, and/or a test of some kind

**Review by m-triassi — APPROVED** [review]
Honestly just one little comment, but i dont think that should hold up a merge

File: src/helpers.php (hunk 1 above)
Comment (a-drew) [reply src/helpers.php:27]: yeah adding tests is a good idea in general. wonder how well we can mock our the actual api clients.

**Comment by a-drew** [comment]
@m-triassi any thoughts on using a helper file in general? `html_split` could live inside the AbstractTranslator class itself but its different enough that I figured we can leave it out. 

+ there's a chance we need to add another thing to handle split sentences / words. I imagine if you str_split and a word ends up in 2 diff chunks, the translation api will mess up that word or if its a sentence the lack of context could make it pick a different word 🤷

**Comment by m-triassi** [comment]
@a-drew I think your intuition is right here, imho, putting this in a helper makes the most sense. Especially now, considering we may have more helper style functions.

**Pull request merged**